csrc/ops/composite.cpp

```diff
@@ -331,12 +335,44 @@ TensorView* matmul(TensorView* tv_a, TensorView* tv_b) {
       " and ",
       tv_b->dtype());
 
+  // Check for K=1 i.e. reduction of broadcast. In these cases we don't need a
```

Comment:
I considered placing this translation to `mul` in another PR, but it is relatively compact and it is difficult to work around if we want to exhaustively check ID mappings with all different combinations of inputs.

Reply:
Thanks for adding these cases!